### DISCUSSION TITLE
CPU utilization decrease

### DISCUSSION THREAD

**Discussion by simon-hans** [kogalur/randomForestSRC#215] [Q&A]
Hello everyone, I am a MD (hematologist) interested in predicting thromboembolic events in patients with cancer, using basic demographics, lab data and somatic genetic information. I have time-to-event data. I need to factor mortality as a competing risk of thrombosis in my models.

On Windows 10, with RStudio and randomForestSRC 3.0.2, if I use a large cohort (>17000) and sample fraction (63%) for about 90 predictors, using 2 outcomes and 1000 trees/forest with rfsrc, speed is good for the first model, but if I fit subsequent models on the exact same cohort the process is slower and slower. Much lower sample sizes (e.g. 1500) do not cause this problem, unless I increase the # of forests substantially (e.g. 10000), then the same thing happens again. I am puzzled that there is "state persistence", i.e. running the same function again and again on the same data results in "accumulative" behavior, as if a buffer was filling etc.

Looking at CPU utilization (24 cores, openmp enabled), for the first fit I get 100% utilization, then for following models this value drops more and more, with a peculiar zig-zag pattern on the plot. The only way to address the problem is to open a new R session. I do not have this problem if I revert to version 2.10.0. I rm() my large model and run gc() every time, so memory usage is under control.

Please let me know what you think, Thank you!

Simon

**Comment by kogalur** [comment]
I think there are two confounding factors here, the first being CPU utilization, and the second being memory utilization.  We need to be able to replicate your scenario before we can determine what is happening.  We have a few questions:

1.  Are you able to provide us with a simple test script that will enable us to duplicate the issue, with either actual or simulated data?
2. Can you confirm that, under otherwise identical conditions, the issue does not occur with 2.10.0, but does occur with 3.0.2?
3. Are you able to provide us with screen shots or video captures of CPU and memory utilization that might help us in analyzing the issue?
4. What is your RAM installed on the machine?

Off-hand, it feels like a memory issue and not a CPU thread related issue.  It feels like objects are building up in the environment, and memory is being paged in and out.  RAM  utilization could be an issue with growing 10K trees in a survival scenario.   We "usually" find 500-1000 trees is adequate.  Watching and reporting on your RAM profile as you do the experiment would be helpful. Thank you
